### PR TITLE
[nnc] Move batchnorm to operators library

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59992 [nnc] Move batchnorm to operators library**
* #59988 [nnc] Move operator implementations into a subdirectory

Wrapped batch norm in function `computeBatchNorm`.

Differential Revision: [D29116661](https://our.internmc.facebook.com/intern/diff/D29116661/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29116661/)!

### CODE REVIEW

**Review by ZolotukhinM — APPROVED** [review]
Looks good assuming it's a verbatim move!
